docs/en/stack/ml/anomaly-detection/ml-revert-model-snapshot.asciidoc

```diff
@@ -0,0 +1,57 @@
+[role="xpack"]
+[[ml-reverting-model-snapshot]]
+= Reverting to a model snapshot
+
+{ml-docs}/ml-ad-run-jobs.html#ml-ad-model-snapshots[Snapshots of the {ml} model] 
+for each {anomaly-job} are saved frequently to an internal {es} index to ensure 
+resilience. It makes it possible to reset the model to a previous state in case 
+of a system failure or if the model changed significantly due to a one-off 
+event.
+
+. In {kib}, navigate to **{ml-app} > Anomaly Detection > Jobs**.
+. Locate the {anomaly-job} whose model you want to revert in the job table.
+. Open the job details and navigate to the **Model Snapshots** tab.
++
+--
+[role="screenshot"]
+image::images/anomaly-job-model-snapshots.jpg[A screenshot of a job with the Model Snapshots tab opened]
+--
+. Select a snapshot from the list and click the **Revert** icon under 
+  **Actions**.
+. Optional: Select if you want to replay the analysis based on the data in your 
+  index after the revert has been applied. If you don’t select this option, 
+  there will be no {anomaly-detect} results after the snapshot was taken and the 
+  job results end at the point in time of the snapshot.If you select this 
+  option, you can then optionally select one or both of the next two options:
+
+  * You can select whether you want the job to continue running in real time 
+  after the replay, or to just replay existing data after the snapshot.
+  * You can select a time range you want to avoid during the replay by declaring 
+  a Calendar event. This way, you can skip any problematic time frame that you 
+  want the {anomaly-job} to avoid.
+
++
+--
+[role="screenshot"]
+image::images/revert-model-snapshot.jpg[A screenshot of a revert model snapshot flyout]
+--
+. Click **Apply**.
+
+TIP: You can use <<ml-ad-rules,custom rules>> to avoid a model being updated in 
+  case of a known event you want to exclude from the analysis. Using custom 
+  rules might help you to avoid situations where you need to revert to a 
+  snapshot.
+
+Alternatively, you can use the 
+{ref}/ml-revert-snapshot.html[Revert model snapshots] API. In this case, you 
+need to manually close the corresponding job before reverting to the saved 
+snapshot.
+
+[NOTE]
+===============================
+* In case of reverting to a snapshot, all {anomaly-detect} results are deleted 
```

Comment:
I found "in case of reverting to a snapshot" unclear.  Which particular case is it referring to? Only when you do it via API?  Or some other circumstance that I'm missing? If it's applicable to all methods, something like this might be clearer:
```suggestion
* By default, when you revert to a snapshot, all {anomaly-detect} results are deleted 
```